torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -857,7 +870,11 @@ struct CodeImpl {
 
   void emitWarn(Node* node) {
     emitLoadInputs(node->inputs());
-    insertInstruction(WARN);
+    int64_t idx = -1;
+    if (node->hasAttribute(attr::warn_id)) {
```

Comment:
just curious - why do we need to have separate indices? would using just `Node*` as a key here be sufficient? or are there some considerations with inlining that might change that? (I'm not TS expert, so @suo probably has a better idea)

Reply:
Good point. Changed to using `Node*` as key.

Reply:
New turn of events.

Using `Node*` as key doesn't work for ProflingExecutor, which unrolls loops and thus creating many copies of same node, resulting in many calls to warnings. Switched back to using an unique ID attached to `aten::warn`.